kDrive/UI/View/Files/Preview/VideoCollectionViewCell.swift

```diff
@@ -125,4 +144,29 @@ class VideoCollectionViewCell: PreviewCollectionViewCell {
             playerViewController.player?.play()
         }
     }
+
+    private func setNowPlayingMetadata() {
```

Comment:
Architecture wise, it is complex enough to justify a `SingleVideoPlayer` struct to encapsulate the media player observation.

You can copy what was done with `SingleTrackPlayer` for the link between the view and the player object.